lib/_stream_readable.js

```diff
@@ -192,9 +192,8 @@ Readable.prototype._destroy = function(err, cb) {
 // This returns true if the highWaterMark has not been hit yet,
 // similar to how Writable.write() returns true if you should
 // write() some more.
-Readable.prototype.push = function(chunk, encoding) {
+Readable.prototype.push = function(chunk, encoding, skipChunkCheck) {
```

Comment:
Not sure if this is acceptable, but I have no better ideas. If it's ok, should we document it or keep it private?

Reply:
Another option is to use a special `encoding` value to specify that `chunk` is a buffer and should not be checked.

Reply:
Not a fan of this approach. What would likely be better is a new internal only `Readable.prototype[kPush] = function(...)` that `Readable.prototype.push(...)` can defer to and that our internal code can call directly.. then keep the type check in the existing `push()` and skip it in the internal one.

Whatever happens here, tho, the @nodejs/streams wg needs to take a look :-)

Reply:
The private `push` would basically be just an helper for https://github.com/nodejs/node/blob/f2e02883e7ef6ff58331e980f91bc31d8076a5c8/lib/_stream_readable.js#L225 so maybe we can add that to the prototype hidden behind a symbol?